rabbitmq/templates/rabbitmq.yaml

```diff
@@ -0,0 +1,205 @@
+        forcePull: true
+        from:
+          kind: ImageStreamTag
+          name: "${FROM_IMAGE}"
+      type: Docker
+    triggers:
+    - imageChange: {}
+      type: ImageChange
+    - type: ConfigChange
+- apiVersion: v1
+  kind: ImageStream
+  metadata:
+    name: centos
+  spec:
+    dockerImageRepository: docker.io/centos
+    lookupPolicy:
+      local: false
+    tags:
+    - from:
+        kind: DockerImage
+        name: docker.io/centos:7
+      name: "7"
+      referencePolicy:
+        type: Source
+    - from:
+        kind: DockerImage
+        name: docker.io/centos:latest
+      name: latest
+      referencePolicy:
+        type: Source
+- apiVersion: v1
+  kind: ImageStream
+  metadata:
+    name: "${APPLICATION_NAME}"
+    labels:
+      application: "${APPLICATION_NAME}"
+- apiVersion: v1
+  kind: ImageStream
+  metadata:
+    name: rhel7
+  spec:
+    dockerImageRepository: registry.access.redhat.com/rhel7/rhel
+    lookupPolicy:
+      local: false
+    tags:
+    - from:
+        kind: DockerImage
+        name: registry.access.redhat.com/rhel7/rhel:7.4
+      name: "7.4"
+      referencePolicy:
+        type: Source
+    - from:
+        kind: DockerImage
+        name: registry.access.redhat.com/rhel7/rhel:latest
+      name: latest
+      referencePolicy:
+        type: Source
+- apiVersion: v1
+  kind: Service
+  metadata:
+    labels:
+      application: "${APPLICATION_NAME}"
+    name: "${APPLICATION_NAME}"
+  spec:
+    clusterIP: None
+    ports:
+    - name: 4369-tcp
+      port: 4369
+      protocol: TCP
+      targetPort: 4369
+    - name: 5671-tcp
+      port: 5671
+      protocol: TCP
+      targetPort: 5671
+    - name: 5672-tcp
+      port: 5672
+      protocol: TCP
+      targetPort: 5672
+    - name: 25672-tcp
+      port: 25672
+      protocol: TCP
+      targetPort: 25672
+    selector:
+      application: "${APPLICATION_NAME}"
+    sessionAffinity: None
+    type: ClusterIP
+- apiVersion: apps/v1beta1
+  kind: StatefulSet
+  metadata:
+    labels:
+      application: "${APPLICATION_NAME}"
+    name: "${APPLICATION_NAME}"
+  spec:
+    replicas: 3
+    selector:
+      matchLabels:
+        application: "${APPLICATION_NAME}"
+    serviceName: "${APPLICATION_NAME}"
+    template:
+      metadata:
+        labels:
+          application: "${APPLICATION_NAME}"
+      spec:
+        containers:
+        - env:
+          - name: AUTOCLUSTER_DELAY
+            value: "10"
+          - name: AUTOCLUSTER_TYPE
+            value: k8s
+          - name: MY_POD_IP
+            valueFrom:
+              fieldRef:
+                apiVersion: v1
+                fieldPath: status.podIP
+          - name: RABBITMQ_ERLANG_COOKIE
+            value: sup3rS3cr1+
+          - name: RABBITMQ_NODENAME
+            value: rabbit@$(MY_POD_IP)
+          - name: RABBITMQ_USE_LONGNAME
+            value: "true"
+          image: "172.30.1.1:5000/${APPLICATION_NAME}/${APPLICATION_NAME}:${RABBITMQ_VERSION}"
```

Comment:
Start in OpenShift 3.6, native Kubernetes resources can make use of ImageStreams

https://docs.openshift.com/container-platform/3.6/release_notes/ocp_3_6_release_notes.html#ocp-36-container-orchestration

It can be enabled by adding the following to the ImageStream and then referencing the name of the imagestream here.
```
  spec:
    lookupPolicy:
      local: true
```